spell/psm-checklists.md

```diff
@@ -0,0 +1,113 @@
+# PSM Checklists
+
+## PSM Migration Checklist
+
+- [ ] IF contracts are being on-boarded in the current spell, execute the full [LitePSM On-boarding Checklist](#litepsm-on-boarding-checklist)
+- [ ] IF a PSM is deprecated in the migration, execute the full [PSM Off-boarding Checklist](#psm-off-boarding-checklist)
+- [ ] Amount of `gem` being migrated matches the Exec Sheet
+- [ ] Collateral amount that is fetched from the source PSM matches the collateral is added to the destination PSM
+- [ ] No bad debt is left behind
+- Source PSM:
+    - [ ] ilk matches source PSM
+    - [ ] IF present, `tin` matches the value in the Exec Sheet
+    - [ ] IF present, `tout` matches the value in the Exec Sheet
+    - [ ] `AutoLine` is updated according to the Exec Sheet
+- Destination PSM:
+    - [ ] ilk matches destination PSM
+    - [ ] IF present, `tin` matches the value in the Exec Sheet
+    - [ ] IF present, `tout` matches the value in the Exec Sheet
+    - [ ] `AutoLine` is updated according to the Exec Sheet
+    - [ ] `buf` matches the value in the Exec Sheet
+    - [ ] `fill` is called to ensure liquidity is immediately available after spell execution
+
+## LitePSM On-boarding Checklist
+
+- Deployed Contracts
+    - `DssLitePsm`
+        - [ ] Uses the agreed best Solidity compiler version at the time
+        - [ ] Optimizer is **enabled**
```

Comment:
Can we fix amount of cycles?

Reply:
> We could enforce this through the checklist, but I guess this is a decision for the author and does not impact the overall security of the contract. In contrast, enforcing the optimizer to be enabled or not does have some implications, and therefore should be in the checklist.

I actually wouldn't say that `--optimize-runs` doesn't have security implications, although being rare, there were optimizer-related bugs (e.g. [one fixed in version 0.8.3](https://soliditylang.org/blog/2021/03/23/keccak-optimizer-bug/)). Each specific `--optimize-runs` might produce differently arranged bytecode which can be manipulated until some vulnerability is present.

Anyway, my initial comment had very basic assumption that being more explicit is always better. If you've already settled with 200 runs, you might want to make it part of the checklist, even if we don't expect that this number will have major security implications.

Reply:
In the case you mentioned, it was not exactly related to `--optimize-runs`, but to the optimizer being enabled in the first place, regardless the number of configured runs.

While I agree that the bytecode might differ depending on the value configured, the LitePSM contract is formally verified with Certora. This ensures correctness at the bytecode level, so as long as the deployed version matches the [configuration in the Certora spec](https://github.com/makerdao/dss-lite-psm/blob/main/certora/DssLitePsm.conf#L26), we have a high degree of confidence that it does not introduce any bugs, regardless of the value of `--optimize-runs`.

In either case, I agree to add the check for 200 runs for explicitness’ sake.